libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineRouteRetrievalTask.java

```diff
@@ -41,7 +53,15 @@ protected DirectionsRoute doInBackground(OfflineRoute... offlineRoutes) {
   @Override
   protected void onPostExecute(DirectionsRoute offlineRoute) {
     if (offlineRoute != null) {
+      routeRetrievalBuilder.end();
```

Comment:
I'm wondering if `AsyncTask` here could further "delay" the timing here for offline route retrieval.  Maybe we should end in the `doInBackground`?  If it's negligible then no worries.